packages/components/Button/src/Button.tsx

```diff
@@ -34,19 +34,56 @@ export const buttonLookup = (layer: string, state: IPressableState, userProps: B
   );
 };
 
+export const extractOuterStylePropsAndroid = memoize((style) => {
+  const marginKeys = [
+    'margin',
+    'marginTop',
+    'marginBottom',
+    'marginLeft',
+    'marginRight',
+    'marginStart',
+    'marginEnd',
+    'marginVertical',
+    'marginHorizontal',
+    'start',
+    'end',
+    'left',
+    'right',
+    'top',
+    'bottom',
+  ];
+  const outerStyleProps = {},
+    resetProps = {};
+  marginKeys.forEach((key) => {
+    if (style && style[key]) {
+      outerStyleProps[key] = style[key];
+      resetProps[key] = 0;
+    }
+  });
+  return [
+    { ...outerStyleProps, display: style ? style.display : 'flex' },
+    { ...style, ...resetProps },
+  ];
+});
+
 export const Button = compose<ButtonType>({
   displayName: buttonName,
   ...stylingSettings,
   slots: {
     root: View,
+    ripple: Pressable,
     icon: Icon,
     content: Text,
   },
   useRender: (userProps: ButtonProps, useSlots: UseSlots<ButtonType>) => {
     const button = useButton(userProps);
+    const [ripplePressedAndroid, setRippleStateAndroid] = React.useState(false);
```

Comment:
I think we can put this in useButton instead of putting it in this file so that all the hooks are in one place.

Reply:
Had done that initially, but then useButton is shared by Compound Button and Toggle Button as well and would unnecessarily impact them. 
Also, based on discussions with @Saadnajmi on his work on migrating to [usePressableState](https://github.com/microsoft/fluentui-react-native/pull/2153), we feel that this hook work around should be a temporary workaround, especially since usePressableState directly allows us to use androidRipple and works better by changing the state (and colors) of button even on a short tap (only on long press color change occurs), while useAsPressable does not do that.